internal/datasource/config/config_writer.go

```diff
@@ -127,32 +132,17 @@ func (cw *ConfigWriter) updateFile(ctx context.Context, fileData *instances.File
 }
 
 func (cw *ConfigWriter) Complete() error {
-	cache, err := json.MarshalIndent(cw.currentFileCache, "", "  ")
+	slog.Debug("Completing config apply")
+	err := cw.fileCache.UpdateFileCache(cw.currentFileCache)
 	if err != nil {
-		return fmt.Errorf("error marshaling cache data from %s: %w", cw.cachePath, err)
+		return fmt.Errorf("error updating cache to %s: %w", cw.fileCache.GetCachePath(), err)
 	}
 
-	err = writeFile(cache, cw.cachePath)
-	if err != nil {
-		return fmt.Errorf("error writing cache to %s: %w", cw.cachePath, err)
-	}
-
-	return err
-}
-
-func (cw *ConfigWriter) SetDataPlaneConfig(dataPlaneConfig config.DataPlaneConfig) {
-	cw.dataPlaneConfig = dataPlaneConfig
-}
-
-func (cw *ConfigWriter) Reload(instance *instances.Instance) error {
-	return cw.dataPlaneConfig.Reload(instance)
-}
-
-func (cw *ConfigWriter) Validate(instance *instances.Instance) error {
-	return cw.dataPlaneConfig.Validate(instance)
+	return nil
 }
 
 func writeFile(fileContent []byte, filePath string) error {
+	slog.Debug("Writing to file", "filePath", filePath)
```

Comment:
```suggestion
	slog.Debug("Writing to file", "file_path", filePath)
```

Reply:
Can I ask why the log should be file_path when the variable is filePath? just a little confused, should this be changed everywhere as I think we are logging the variable names everywhere else

Reply:
I was getting a linter warning in my IDE complaining that it should be snake_case. Dunno if that's a common convention for apps that consume logs, such as Logstash.

Not a big deal either way, you can leave it as is if you want.